Sources/VLCMovieViewController.m

```diff
@@ -564,6 +564,8 @@ - (void)viewWillDisappear:(BOOL)animated
 
 - (void)viewDidDisappear:(BOOL)animated
 {
+    // reset toggleButton to default icon when movieviewcontroller is dismissed
+    _videoOptionsControlBar.isInFullScreen = NO;
```

Comment:
that leads to a wrong state, if you go into fullscreen and switch to miniplayer and back to maxiplayer the button has the wrong state because of this

Reply:
I would set that in prepareForMediaPlayback which is only called when a new media is started

Reply:
Actually, prepareForMediaPlayback is called everytime the view comes back into view. But I didnt know that there was a playbackDidStop notification, which is better for resetting I believe.